# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsWindow.cpp
# Commit: 427273080257
# Full Hash: 42727308025767aa9f53eaf149b13b154f537bc8
# Author: Botond Ballo <botond@mozilla.com>
# Date: 2019-10-01 04:16:24
# Regressor Bug: 1511901
# File Overlap Count: 1
# Description:
#   Bug 1511901 - Fix device name comparison in TouchDeviceNeedsPanGestureConversion(). r=cmartin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45113
# ==============================================================================

diff -r df723e7120dc -r 427273080257 widget/windows/nsWindow.cpp
--- a/widget/windows/nsWindow.cpp	Mon Sep 30 17:48:34 2019 +0000
+++ b/widget/windows/nsWindow.cpp	Mon Sep 30 19:43:35 2019 +0000
@@ -6724,7 +6724,14 @@
   }
   // The affected device name is "\\?\VIRTUAL_DIGITIZER", but each backslash
   // needs to be escaped with another one.
-  if (deviceName != "\\\\?\\VIRTUAL_DIGITIZER") {
+  std::string expectedDeviceName = "\\\\?\\VIRTUAL_DIGITIZER";
+  // For some reason, the dataSize returned by the first call is double the
+  // actual length of the device name (as if it were returning the size of a
+  // wide-character string in bytes) even though we are using the narrow
+  // version of the API. For the comparison against the expected device name
+  // to pass, we truncate the buffer to be no longer tha the expected device
+  // name.
+  if (deviceName.substr(0, expectedDeviceName.length()) != expectedDeviceName) {
     return false;
   }
 
